### PR TITLE
Update 'range' functionality

### PR DESCRIPTION
**Pull request by pollackscience**
"range" kw now masks data and and is never ignored.  This impacts automatic binning methods, especially bayesian blocks.  These methods will not consider data outside of user-defined range when determining bin edges.

### CODE REVIEW

**Comment by pollackscience** [comment]
A test is failing in Travis CI that is part of a different module: `tests/dataset/test_selection.py` fails for 3 of 4 python versions. @eduardo-rodrigues not sure who to flag about this.

File: .project [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/brianpollack/Coding/scikit-hep
```
Comment (eduardo-rodrigues) [inline .project]: Can you remove this file and add a line in .gitignore so that such files are hereafter ignored :-)?

**Comment by eduardo-rodrigues** [comment]
I'm cc'ing @marinang as his the dataset expect and largest author ...

File: .project (hunk 1 above)
Comment (pollackscience) [reply .project]: whoops, of course

**Comment by marinang** [comment]
mmh this is strange... looks like a pyparsing issue returning a different format in selection.

**Comment by eduardo-rodrigues** [comment]
@pollackscience, there's some discussion on the PyHEP Gitter channel that should be of interest to you. I've just mentioned your work in this package but could not ping you as I believe you are not on the channel, see https://gitter.im/HSF/PyHEP. This is actually the means of discussion of the HSF "Python in HEP" WG, see https://hepsoftwarefoundation.org/workinggroups/pyhep.html. You are more than welcome to contribute :-).

**Comment by marinang** [comment]
I see that the only difference between the failing and succeeding builds in the pyparsing version. So Maybe a quick fix is to require pyparsing version <= 2.2.0.

**Comment by marinang** [comment]
I have found a fix working with both versions of pyparsing maybe I can append it here?

**Comment by eduardo-rodrigues** [comment]
Sure, @marinang, append it. Thanks!

**Comment by marinang** [comment]
@eduardo-rodrigues actually I cannot add commit to this pull request.

**Comment by eduardo-rodrigues** [comment]
Oh, you would need to fork @pollackscience's clone and commit there … If that's too much trouble I suggest you open a new PR and that gets merged beforehand. Then we can rebase, if needed, or else retrigger the CI to see things OK again ...

**Comment by marinang** [comment]
@eduardo-rodrigues sounds better to do a PR then.

**Comment by pollackscience** [comment]
@eduardo-rodrigues @marinang  Sorry for the delay, I can throw in the change to my commit if you'd like and just push it, just let me know what to modify.

**Comment by eduardo-rodrigues** [comment]
FYI @pollackscience, I've just retriggered Travis given the fixes from @marinang with https://github.com/scikit-hep/scikit-hep/pull/76.

**Comment by pollackscience** [comment]
Yup, just merged those changes in and pushed my PR

**Comment by pollackscience** [comment]
@eduardo-rodrigues looks like this should be ready to merge.

**Comment by eduardo-rodrigues** [comment]
Have been on a long weekend ;-). Merging, thanks a lot!

**Pull request merged**